httpclient5/src/test/java/org/apache/hc/client5/http/impl/auth/TestDigestScheme.java

```diff
@@ -746,4 +746,25 @@ public void testSerialization() throws Exception {
         Assertions.assertEquals(digestScheme.getCnonce(), authScheme.getCnonce());
     }
 
+
+    @Test
+    public void testErrorOnBothUsernameAndUsernameStar() throws Exception {
+        final ClassicHttpRequest request = new BasicClassicHttpRequest("Post", "/");
+        request.setEntity(new InputStreamEntity(new ByteArrayInputStream(new byte[] {'a'}), -1, ContentType.DEFAULT_TEXT));
+        final HttpHost host = new HttpHost("somehost", 80);
+        final CredentialsProvider credentialsProvider = CredentialsProviderBuilder.create()
+                .add(new AuthScope(host, "realm1", null), "username", "password".toCharArray())
+                .build();
+
+        final String challenge = StandardAuthScheme.DIGEST + " realm=\"realm1\", nonce=\"f2a3f18799759d4f1a1c068b92b573cb\", " +
```

Comment:
@arturobernalg Please note this scenario is not very practical. The `username` and `username*` parameters apply to an auth response, not to an auth challenge. 
I approved the change-set but in all honesty the check is not very useful.

Reply:
> @ok2c, I understand the practicality aspect now. Could you please advise on how best to handle this test to ensure it aligns more closely with common scenarios involving the username and username* parameters? Any suggestions or examples would be greatly appreciated.

@arturobernalg This check makes sense on the server side, which we have no support for. It makes little sense on the client side. I would drop the change-set altogether.